config/config.go

```diff
@@ -364,13 +367,20 @@ func ParseUpstream(upstream string) (Upstream, error) {
 	}
 
 	return Upstream{
-		Net:  n,
-		Host: host,
-		Port: port,
-		Path: path,
+		Net:        n,
+		Host:       host,
+		Port:       port,
+		Path:       path,
+		CommonName: commonName,
 	}, nil
 }
 
+func extractCN(in string) (cn string, upstream string) {
+	upstream, cn, _ = strings.Cut(in, "#")
+
+	return upstream, cn
```

Comment:
Small typo: forgot to inverse the returned vars.
```suggestion
	return cn, upstream
```